keras/backend.py

```diff
@@ -5547,6 +5559,14 @@ def sparse_categorical_crossentropy(target, output, from_logits=False, axis=-1):
         axis: Int specifying the channels axis. `axis=-1` corresponds to data
             format `channels_last`, and `axis=1` corresponds to data format
             `channels_first`.
+        ignore_index: Optional integer, the id of a label that will not be
+            included in the entropy equation nor in gradient computation. This
+            is useful in segmentation problems containing the *void* label
+            (commonly -1 or 255) in its annotated segmentation maps.
+            By default, all label ids are considered. If `ignore_index` is not
+            `None` and the output is a tensor with `rank>=3`, then the valid
+            entries will be averaged over the axes `range(1, output_rank-1)`,
+            resulting in an output of shape `[batch]`.
```

Comment:
```
            Optional integer in the range `0 <= ignore_class < num_classes`.
            The ID of a class to be ignored during loss computation.
            This is useful, for example, in segmentation problems featuring a "void" class
            (commonly -1 or 255) in segmentation maps.
            By default (`ignore_class=None`), all classes are considered.
```

Reply:
Okay, but `ignore_class` is not constrained to the `[0, num_classes)` interval. For example, [DeepLab](https://github.com/tensorflow/models/blob/d81c7d4b35404bae2a7f3583668fdd1cfc51fce5/research/deeplab/datasets/data_generator.py#L79) and many other segmentation repos use `ignore_class == 255 > num_classes`.
Also, you scratched the reduction part, but people need to know that `shape(crossentropy(y, p)) == [batch, height, width]` and `shape(crossentropy(y, p, ignore_class=-1) == [batch]`, so they don't make assumptions that will lead to incorrect operations.

Reply:
> Also, you scratched the reduction part, but people need to know that shape(crossentropy(y, p)) == [batch, height, width] and shape(crossentropy(y, p, ignore_class=-1) == [batch], so they don't make assumptions that will lead to incorrect operations.

I had miss this. Why is that? It seems like a big inconsistency. Ignoring a class should not change the reduction behavior, no?

Reply:
I answered [directly in the PR](https://github.com/keras-team/keras/pull/16712#issuecomment-1174507974) because I thought it'd more comfortable for you to read.